src/cmake/externalpackages.cmake

```
@@ -147,8 +147,8 @@ if (LLVM_VERSION VERSION_GREATER_EQUAL 15.0
          "If you are using LLVM 15 or higher, you should also use clang version "
          "15 or higher, or you may get build errors.${ColorReset}\n")
 endif ()
-if (LLVM_VERSION VERSION_GREATER_EQUAL 16.0)
-    message (ERROR "${ColorYellow}OSL is not yet compatible with LLVM 16.${ColorReset}\n")
+if (LLVM_VERSION VERSION_GREATER_EQUAL 17.0)
+    message (ERROR "${ColorYellow}OSL is not yet compatible with LLVM 17.${ColorReset}\n")
```

```suggestion
if (LLVM_VERSION VERSION_GREATER_EQUAL 17.0)
    message (ERROR "${ColorYellow}OSL is not yet compatible with LLVM 17.${ColorReset}\n")
endif ()
if (LLVM_VERSION VERSION_GREATER_EQUAL 16.0)
```

You inadvertently changed the logic of the following tests and warnings, which SHOULD apply for LLVM 16.